src/amo/components/CollectionManager/index.js

```diff
@@ -57,22 +59,39 @@ export class CollectionManagerBase extends React.Component<Props, State> {
     }
   }
 
-  onCancel = () => {
-    const { errorHandler } = this.props;
+  onCancel = (event: SyntheticEvent<any>) => {
+    const {
+      collection, clientApp, errorHandler, router, siteLang,
+    } = this.props;
+    event.preventDefault();
+    event.stopPropagation();
+    invariant(collection,
+      'A collection must be loaded before you can cancel');
+    invariant(clientApp,
+      'A clientApp must be loaded before you can cancel');
+    invariant(siteLang,
+      'A siteLang must be loaded before you can cancel');
 
     // Reset form state to the original collection object.
     this.setState(this.propsToState(this.props));
     errorHandler.clear();
+
+    const { authorUsername, slug } = collection;
+    router.push(
+      `/${siteLang}/${clientApp}/collections/${authorUsername}/${slug}/`
+    );
   }
 
-  onSubmit = () => {
+  onSubmit = (event: SyntheticEvent<any>) => {
     const {
       collection,
       errorHandler,
       dispatch,
       i18n,
       siteLang,
     } = this.props;
+    event.preventDefault();
+    event.stopPropagation();
 
     if (!collection) {
```

Comment:
I'm guessing though, this error will bubble up to the `errorHandler` and be displayed to the user? If so, leaving a plain-english version (rather than the version invariant will create in production which will have no message) might be handy. 🤷‍♂️

Reply:
I actually wasn't clear on that, and discussed it with Kumar in another PR but I don't think it was resolved. From what I can tell from the docs and the code, unless we configure Webpack to strip the messages in production (which we have not done), `invariant` won't strip them itself. So if we leave everything as is, for now, the same messages should be displayed in prod. Maybe I'm wrong about that, but not as far as I can tell.

Reply:
Either way, *eventually* we should strip them as they're largely for development use. It's probably good to think of them as not accessible in production, where we wanna save all those precious bytes 😉

Reply:
Fair enough. :)